### DISCUSSION TITLE
Is there a way to force DAG file's reload from UI?

### DISCUSSION THREAD

**Discussion by kalyanr23** [apache/airflow#29398] [Q&A] [answered]
Hi all,
Is there a way to force scheduler to reload DAG files from UI? I understand there's a env variable that decides how often scheduler should reload. But, that's a setting I don't have access to update. So, looking for an alternative to quickly reload the changes i made.

Also, is there any info on UI where I can look when Scheduler has last reloaded DAG's?

**Comment by josh-fell** [comment, marked as answer]
Hi @kalyanr23!

For forcing reparsing of the DAGs, you can check out if running [`airflow dags reserialize`](https://airflow.apache.org/docs/apache-airflow/stable/cli-and-env-variables-ref.html#reserialize) CLI command works for your needs. You can use the `--subdir` option to specify individual DAG files to reserialize too.

As far as info in the UI when DAGs were last reloaded/parsed, there isn't one place to see when every DAG in your environment was parsed, but in the DAG Details page there is a `last_parsed_time` attribute. You can at least see where a DAG was last reloaded/parsed for some relative timing and/or freshness of the DAG code generally in your environment.

**Comment by hussein-awala** [comment]
If the command `airflow dags reserialize` proposed by @josh-fell works for your needs, you can create a custom [plugin](https://airflow.apache.org/docs/apache-airflow/stable/authoring-and-scheduling/plugins.html#plugins) to execute the command from the UI.

**Reply by kalyanr23** [reply]
@hussein-awala , we can add new UI components as part of custom plugins?

**Reply by potiuk** [reply]
Yes read the docs (look for plugins)
